vxquery-core/src/main/java/org/apache/vxquery/runtime/functions/json/JnKeysScalarEvaluator.java

```diff
@@ -66,9 +68,10 @@ protected void evaluate(TaggedValuePointable[] args, IPointable result) throws S
                 for (int i = 0; i < size1; i++) {
                     sp1.getEntry(i, temptvp);
                     if (temptvp.getTag() == ValueTag.OBJECT_TAG) {
-                        op = (ObjectPointable) ObjectPointable.FACTORY.createPointable();
                         temptvp.getValue(op);
-                        op.getKeys(temptvp);
+                        abvs1 = new ArrayBackedValueStorage();
```

Comment:
It would be good to be able to pool the `ArrayBackedValueStorage`s for the evaluator. If we use this evaluator on a big dataset, we'll create and collect the  `ArrayBackedValueStorage`s all the time. I'm not sure if there are other cases in which we already pool those, but I think that it would be very valuable here.